DiscordBot/Commands/EventCommands.cs

```diff
@@ -217,6 +219,72 @@ public async Task EditEvent(CommandContext context, InteractivityModule interact
             await EditEventField(context, interactivity, eventKey.Value, editField, eventEmbed);
         }
 
+        private static async Task CreateSignupSheet(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the event key? (use the ``list`` option to find out)");
+            var eventKey = await GetUserIntResponse(context, interactivity);
+            if (eventKey == null)
+            {
+                return;
+            }
+
+            var eventEmbed = await GetEventEmbed(context, eventKey.Value);
+            if (eventEmbed == null)
+            {
+                return;
+            }
+
+            await context.RespondAsync($"{context.Member.Mention} - start signups for this event? - (``yes``/``no``)", embed: eventEmbed);
+            var confirmationResponse = await GetUserConfirmation(context, interactivity);
+            if (confirmationResponse == null || confirmationResponse == false)
+            {
+                return;
+            }
+
+            await SendSignupMessage(context, eventKey.Value);
+        }
+
+        private static async Task SendSignupMessage(CommandContext context, int eventKey)
+        { 
+            var eventsSheetsService = context.Dependencies.GetDependency<IEventsSheetsService>();
+            var discordEvent = await eventsSheetsService.GetEventAsync(eventKey);
+
+            var signupEmbed = new DiscordEmbedBuilder
+            {
+                Title = $"{discordEvent.Name} - {discordEvent.Time:ddd dd MMM yyyy @ h:mm tt}",
+                Description = discordEvent.Description,
+                Footer = new DiscordEmbedBuilder.EmbedFooter
+                {
+                    Text = $"event key: {eventKey}"
+                }
+            };
+
+            var userResponseDictionary = await eventsSheetsService.GetSignupsByUserAsync(eventKey);
+
+            var usersField = string.Join("\n", userResponseDictionary.Select(
+                response => $"<@{response.Key}>"));
+            signupEmbed.AddField("Participants", usersField, true);
+
+            var responsesList = userResponseDictionary.Select(
+                response => string.Join(" ", response.Value));
+            var responsesField = string.Join("\n", responsesList);
+            signupEmbed.AddField("Responses", responsesField, true);
+
+            var optionsList = await eventsSheetsService.GetSignupsByResponseAsync(eventKey);
+            var optionsField = string.Join(
+                "\n", optionsList.Select(response =>
+                    $"{response.Key.Emoji} - {response.Key.ResponseName}"));
```

Comment:
I think it's more readable with the closing brackets on a new line, especially in cases like this 🙂